src/main/java/langley/tasks/Task.java

```diff
@@ -0,0 +1,75 @@
+package langley.tasks;
+
+public abstract class Task {
+    String description;
+    boolean isMarked;
+
+    public Task(String description) {
+        this.description = description;
+        this.isMarked = false;
+    }
+
+    /**
+     * Marks the task.
+     */
+    public void mark() {
+        this.isMarked = true;
+    }
+
+    /**
+     * Unmarks the task.
+     */
+    public void unmark() {
+        this.isMarked = false;
+    }
+
+    public abstract String getType();
+
+    @Override
+    public String toString() {
+        return "[" + getType() + "][" + (isMarked ? "X" : " ") + "] " + description;
+    }
+
+    /**
+     * Converts a string representation of a task back into a Task object.
+     * Parses the string for type, status, and task details.
+     *
+     * @param line The string representing a task.
+     * @return The corresponding Task object or null if parsing fails.
+     */
+    public static Task fromString(String line) {
+        String[] parts = line.split("]", 3);
+        String type = parts[0].substring(1);
+        boolean isMarked = parts[1].charAt(1) == 'X';
+        String description = parts[2].trim();
+
+        Task task = null;
+
+        switch (type) {
```

Comment:
The switch statement should have the following form: Note there is no indentation for case clauses.

```java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
} 
```